### ISSUE TITLE
Subfolders Support

### ISSUE THREAD

**Issue by glegenty** [cocopon/tweakpane#27]
Hi ! 
Do you plan to handle subfolders ?

**Comment by cocopon** [comment]
Currently I don't have a plan but I'm interested in the feature. Do you have any specific examples that can be shared?

**Comment by glegenty** [comment]
For example when i'm working on a three.js scene i could have : 

> Main-Pane
| - - -> Camera (folder)
|	| - -	> Toggle (function)
|	| - -	> Position (folder)
|		| - > x (slider)
|		| - > y (slider)	
|		| - > z (slider)
| - - -> 3D Object (folder)
	| - -	> Explode (function)
	| - -	> Position (folder)
	|	| - > x (slider)
	|	| - > y (slider)	
	|	| - > z (slider)	
	| - -	> Rotation (folder)
		| - > x (slider)
		| - > y (slider)	
		| - > z (slider)

**Comment by glegenty** [comment]
For example when i'm working on three.js scene i could have :

\> Main-Pane
| - - -> Camera (folder)
|&nbsp;&nbsp;&nbsp;&nbsp;| - - > Toggle (function)
|&nbsp;&nbsp;&nbsp;&nbsp;| - - > Position (folder)
|&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| - > x (slider)
|&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| - > y (slider)&nbsp;&nbsp;&nbsp;&nbsp;
|&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| - > z (slider)
| - - -> 3D Object (folder)
|&nbsp;&nbsp;&nbsp;&nbsp;| - - > Toggle (function)
|&nbsp;&nbsp;&nbsp;&nbsp;| - - > Position (folder)
|&nbsp;&nbsp;&nbsp;&nbsp;|&nbsp;&nbsp;&nbsp;&nbsp;| - > x (slider)
|&nbsp;&nbsp;&nbsp;&nbsp;|&nbsp;&nbsp;&nbsp;&nbsp;| - > y (slider)&nbsp;&nbsp;&nbsp;&nbsp;
|&nbsp;&nbsp;&nbsp;&nbsp;|&nbsp;&nbsp;&nbsp;&nbsp;| - > z (slider)&nbsp;&nbsp;&nbsp;&nbsp;
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| - - > Rotation (folder)
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| - > x (slider)
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| - > y (slider)&nbsp;&nbsp;&nbsp;&nbsp;
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;| - > z (slider)&nbsp;&nbsp;&nbsp;&nbsp;

**Comment by lavolpecheprogramma** [comment]
I'm interested too and thanks for your work!

**Comment by charlieroberts** [comment]
great library! I agree with this, I can't see a way to represent non-trivial 3D scenes without subfolders...

**Comment by miltoncandelero** [comment]
Another (similar) usecase:

To represent the Stage of pixijs:

```
Stage
| -> Sprite (folder)
|      | -> Position (x,y)
|      | -> Scale (x,y)
| -> Container (folder)
|      | -> Position (x,y)
|      | -> Scale (x,y)
|      | -> Children (folder)
|      |      | -> Sprite (folder)
|      |             | -> Position (x,y)
|      |             | -> Scale (x,y)
|      | -> Container (folder)
|      |      | -> Position (x,y)
|      |      | -> Scale (x,y)
|      |      | -> Children (folder)
|      |             | -> Sprite (folder)
|      |             |      | -> Position (x,y)
|      |             |      | -> Scale (x,y)
|      |             | -> Sprite (folder)
|      |                    | -> Position (x,y)
|      |                    | -> Scale (x,y)
|      | -> Sprite (folder)
|             | -> Position (x,y)
|             | -> Scale (x,y)
| -> ...
```

**Comment by cocopon** [comment]
Thank you for requesting the feature. Released in [1.4.0](https://github.com/cocopon/tweakpane/releases/tag/1.4.0)!
